File: frontend/src/app/core/services/review.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { PageResponse } from '../models/api.model';
import { Review, ReviewCreateRequest } from '../models/review.model';

@Injectable({ providedIn: 'root' })
export class ReviewService {
  private api = inject(ApiService);

  createReview(contractId: number, req: ReviewCreateRequest): Observable<Review> {
    return this.api.post<Review>(`/contracts/${contractId}/review`, req);
  }

  getReviewsByUser(userId: number, page = 0, size = 10): Observable<PageResponse<Review>> {
    return this.api.get<PageResponse<Review>>(`/users/${userId}/reviews`, { page, size });
  }
}

File: backend/src/main/java/com/freelancer/service/ReviewService.java
package com.freelancer.service;

import com.freelancer.dto.request.ReviewCreateRequest;
import com.freelancer.dto.response.ReviewResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ReviewService {

    ReviewResponse createReview(Long contractId, ReviewCreateRequest req, Long reviewerId);

    Page<ReviewResponse> getReviewsByUser(Long userId, Pageable pageable);
}
